evalml/pipelines/pipeline_base.py

```diff
@@ -432,38 +434,51 @@ def graph_json(self):
         y_edges specifies from which component target data is being passed.
         This can be used to build graphs across a variety of visualization tools.
         Template:
-        {"Nodes": {"component_name": {"Name": class_name, "Attributes": parameters_attributes}, ...}},
+        {"Nodes": {"component_name": {"Name": class_name, "Parameters": parameters_attributes}, ...}},
         "x_edges": [[from_component_name, to_component_name], [from_component_name, to_component_name], ...],
         "y_edges": [[from_component_name, to_component_name], [from_component_name, to_component_name], ...]}
 
         Returns:
             dag_json (str): A serialized JSON representation of a DAG structure.
         """
-        nodes = {
-            comp_: {"Attributes": att_.parameters, "Name": att_.name}
-            for comp_, att_ in self.component_graph.component_instances.items()
-        }
-
-        x_edges = self.component_graph._get_edges(
+        nodes = {}
+        for comp_, att_ in self.component_graph.component_instances.items():
```

Comment:
Just a thought but since we do something like this in LG, would overwriting the default encoder help us here? It's very similar to the impl you have here where we check the value of the attribute but it keeps all of the checking of attribute isolated in an object:
```
def dict_to_json_str(dict_in, indent=4):
    """Given a dictionary of values, convert to a JSON string. Safely handles np types as well.

    Args:
        dict_in (dict): the dictionary to be serialized.
        indent (int): the string indent to use on each newline.

    Returns:
        str: the dict as JSON.
    """

    class NpEncoder(json.JSONEncoder):
        def default(self, obj):
            if isinstance(obj, np.integer):
                return int(obj)
            elif isinstance(obj, np.floating):
                return float(obj)
            elif isinstance(obj, np.ndarray):
                return obj.tolist()
            elif isinstance(obj, PipelineBase) or isinstance(obj, ComponentBase):
                return obj.parameters
            try:
                return super(NpEncoder, self).default(obj)
            except TypeError as e:
                logger.warn(f"TypeError using the default encoder: {str(e)}; Will use string representation instead.")
                return str(obj)
    return json.dumps(dict_in, indent=indent, cls=NpEncoder)
```

Reply:
This is very interesting, I didn't know it existed! This also could tie in with what @freddyaboulton was talking about as well. It feels out of scope for this specific PR, but I'll file a new issue to track discussion around hardening this method and/or returning a dict instead.